src/core/blueprint/blueprint.rs

```diff
@@ -77,6 +77,15 @@ impl Type {
             Type::ListType { non_null, .. } => *non_null,
         }
     }
+
+    // checks if inner type is nullable
+    pub fn is_inner_nullable(&self) -> bool {
+        match self {
+            Type::NamedType { non_null, .. } => !*non_null,
+            Type::ListType { of_type, .. } => of_type.is_nullable(),
+        }
+    }
```

Comment:
Technically it should return an `Option<bool>` inner isn't available if self is a NamedType. I would rather have this function inlined into its only usage as this seems more like business logic than a helper method on Type.
Action: Inline the function wherever it's used.